### PR TITLE
Use Installer Type Conforming Installation Directory for Ubuntu

### PR DESCRIPTION
**Pull request by nagilson**
For https://github.com/dotnet/vscode-dotnet-runtime/issues/1520
The /bin/ folder is a symlink that gets used by both distro support types, PMC and Distro Built In.
I got confirmation from Cannonical that lib is where they always put their files and we should expect our PMC debs to place them into share. This is a much simpler change that reading the symlink.

I think I put bin there when testing because thats what I saw the OS doing but it was in reality a symlink.

### CODE REVIEW

**Comment by nagilson** [comment]
This is not quite good enough because it will still think BIN is a custom folder, it needs to resolve the symlink.

File: vscode-dotnet-runtime-library/src/Acquisition/GenericDistroSDKProvider.ts [hunk 1]
```diff
@@ -37,6 +39,18 @@ export class GenericDistroSDKProvider extends IDistroDotnetSDKProvider
         {
             commandResult[0] = commandResult[0].trim();
         }
+
+        if(commandResult && this.resolvePathAsSymlink)
+        {
+            let symLinkReadCommand = this.myDistroCommands(this.readSymbolicLinkCommandKey);
+            symLinkReadCommand = CommandExecutor.replaceSubstringsInCommands(symLinkReadCommand, this.missingPathKey, commandResult[0]);
```
Comment (JL03-Yue) [inline vscode-dotnet-runtime-library/src/Acquisition/GenericDistroSDKProvider.ts:46]: Looks good to me. Could you explain the `missPathKey` variable here?

File: vscode-dotnet-runtime-library/src/test/unit/LinuxDistroTests.test.ts [hunk 2]
```diff
@@ -104,7 +104,7 @@ Microsoft.NETCore.App 7.0.5 [/usr/lib/dotnet/shared/Microsoft.NETCore.App]`;
         if(shouldRun)
         {
             await provider.getInstalledGlobalDotnetPathIfExists(installType);
-            assert.equal(mockExecutor.attemptedCommand, 'which dotnet');
+            assert.equal(mockExecutor.attemptedCommand, 'readlink -f /usr/bin/dotnet');
```
Comment (JL03-Yue) [inline vscode-dotnet-runtime-library/src/test/unit/LinuxDistroTests.test.ts:107]: Thanks for the explanation for the tests!

**Review by JL03-Yue — APPROVED** [review]

File: vscode-dotnet-runtime-library/src/Acquisition/GenericDistroSDKProvider.ts (hunk 1 above)
Comment (nagilson) [reply vscode-dotnet-runtime-library/src/Acquisition/GenericDistroSDKProvider.ts:46]: ah yes, the missing Path Key is the string {path} that gets resolved to become the path that needs to be converted from a symlink to the true path! Thank you for your review :)

**Pull request merged**